ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Util/AnchoredFragment.hs

```diff
@@ -120,7 +120,7 @@ compareAnchoredFragments cfg frag1 frag2 =
       = return ()
       | otherwise
       = throwError
-          "precondition violated: fragments both empty or don't intersect"
+          "precondition violated: fragments aren't both non-empty or don't intersect"
```

Comment:
It should be "and" instead of "or" I think:
```suggestion
          "precondition violated: fragments aren't both non-empty and don't intersect"
````
or alternatively "one of the fragments is empty, and they don't intersect", which is the negation of the precondition "both fragments are non-empty or they intersect." via de Morgan:
```
    negate("both fragments are non-empty or they intersect.")
  ⇔ ¬((¬(null(frag1)) ∧ ¬(null(frag2))) ∨ intersect(frag1,frag2))
  ⇔ ¬(¬(null(frag1)) ∧ ¬(null(frag2))) ∧ ¬(intersect(frag1,frag2))
  ⇔ (null(frag1) ∨ null(frag2)) ∧ ¬(intersect(frag1,frag2))
  ⇔ "one of the fragments is empty, and they don't intersect"
```